nomos/raw/p2p-network.md

```diff
@@ -0,0 +1,278 @@
+
+- Tailored solutions based on user network configuration
+- Automatic NAT type detection and adaptation
+- Fallback mechanisms for challenging network environments
+
+*Note: Detailed NAT traversal specifications are maintained in a separate document.*
+
+### Message Dissemination
+
+#### Gossipsub Protocol
+
+Nomos employs **gossipsub** for reliable message propagation across the network.
+
+**Integration:**
+
+- Seamless integration with Kademlia peer discovery
+- Automatic peer list updates
+- Efficient message routing and delivery
+
+#### Topic Configuration
+
+**Mempool Dissemination:**
+
+- **Mainnet**: `/nomos/mempool/0.1.0`
+- **Testnet**: `/nomos-testnet/mempool/0.1.0`
+
+**Block Propagation:**
+
+- **Mainnet**: `/nomos/cryptarchia/0.1.0`
+- **Testnet**: `/nomos-testnet/cryptarchia/0.1.0`
+
+#### Network Parameters
+
+**Peering Degree:**
+
+- **Minimum recommended**: 8 peers
+- **Rationale**: Ensures redundancy and efficient propagation
+- **Configurable**: Nodes may adjust based on resources and requirements
+
+### Bootstrapping
+
+#### Initial Network Entry
+
+New nodes connect to the network through designated bootstrap nodes.
+
+**Process:**
+
+1. Connect to known bootstrap nodes
+2. Obtain initial peer list through Kademlia
+3. Establish gossipsub connections
+4. Begin participating in network protocols
+
+**Bootstrap Node Requirements:**
+
+- High availability and reliability
+- Geographic distribution
+- Version compatibility maintenance
+
+### Message Encoding
+
+All network messages follow the Nomos Wire Format specification for consistent encoding and decoding across implementations.
+
+**Key Properties:**
+
+- Deterministic serialization
+- Efficient binary encoding
+- Forward/backward compatibility support
+- Cross-platform consistency
+
+*Note: Detailed wire format specifications are maintained in a separate document.*
+
+## Implementation Requirements
+
+### Mandatory Protocols
+
+All Nomos nodes MUST implement:
+
+1. **QUIC transport**
+2. **Kademlia DHT** for peer discovery
+3. **Identify protocol** for peer information exchange
+4. **Gossipsub** for message dissemination
+
+### Optional Enhancements
+
+Nodes MAY implement:
+
+- Advanced NAT traversal techniques
+- Custom peering strategies
+- Enhanced message routing optimizations
+
+### Network Versioning
+
+Protocol versions follow semantic versioning:
+
+- **Major version**: Breaking protocol changes
+- **Minor version**: Backward-compatible enhancements
+- **Patch version**: Bug fixes and optimizations
+
+### Configuration Parameters
+
+#### Peer Discovery Parameters
+
+```yaml
+kademlia:
+  protocol_id: "/nomos/kad/1.0.0"
+  replication_factor: 20
+  query_timeout: 60s
+
+identify:
+  protocol_id: "/nomos/identify/1.0.0"
+  push_interval: 300s
+```
+
+#### Gossipsub Parameters
+
+```yaml
+gossipsub:
+  min_peers: 8
+  max_peers: 200
+  heartbeat_interval: 1s
+  fanout_ttl: 60s
+  topics:
+    - "/nomos/mempool/0.1.0"
+    - "/nomos/cryptarchia/0.1.0"
+```
```

Comment:
This should be checked against the main repo config. Not sure it is correct here.